cryptoki/src/session/object_management.rs

```diff
@@ -96,6 +96,40 @@ impl Session {
         }
     }
 
+    /// Copy an object
+    /// A template can be provided to change some attributes of the new object, when allowed.
+    ///
+    /// # Arguments
+    ///
+    /// * `object` - The [ObjectHandle] used to reference the object to copy
+    /// * `template` - a reference to a slice of attributes
+    ///
+    /// # Returns
+    ///
+    /// This function will return a new [ObjectHandle] that references the newly created object.
+    ///
+    pub fn copy_object(
+        &self,
+        object: ObjectHandle,
+        template: &[Attribute],
+    ) -> Result<ObjectHandle> {
+        let mut template: Vec<CK_ATTRIBUTE> = template.iter().map(|attr| attr.into()).collect();
+        let mut object_handle = 0;
+
+        unsafe {
+            Rv::from(get_pkcs11!(self.client(), C_CopyObject)(
+                self.handle(),
+                object.handle(),
+                template.as_mut_ptr(),
+                template.len().try_into()?,
+                &mut object_handle as CK_OBJECT_HANDLE_PTR,
+            ))
+            .into_result(Function::CreateObject)?;
```

Comment:
I think we need `CopyObject` here:

```suggestion
            .into_result(Function::CopyObject)?;
```

(this may need further changes in `Function` to introduce new variant if it's not there)

Reply:
good catch!